website/pages/docs/k8s/crds/upgrade-to-crds.mdx

```diff
@@ -0,0 +1,318 @@
+          "local_connect_timeout_ms": 1000
+        }
+  ```
+
+## Migration Overview
+
+Overall our migration process will consist of identifying which [config entries](/docs/agent/config-entries)
+have been created in Consul and adding metadata to them so that they can
+be managed by a custom resource instead.
+
+## Why Migrate
+
+All of the above settings do not support modification after the initial
+installation of Consul, i.e. they can not be changed through the Helm chart.
+
+By switching to custom resources, these settings can now be modified.
+
+## Default Protocol
+
+If you are setting `connectInject.centralConfig.defaultProtocol` then you must
+perform the follow steps to migrate to custom resources.
+
+1. Find existing `service-defaults` config entries:
+   ```shell-session
+   $ consul config list -kind service-defaults
+   static-client
+   static-server
+   ```
+1. For each entry, export the config to a file:
+
+   ```shell-session
+   $ consul config read -name static-client -kind service-defaults > static-client.json
+   ```
+
+1. Edit the file and add the key `"Meta": {"consul.hashicorp.com/source-datacenter": "dc1"}`.
+   Where `dc1` is the name of your datacenter. Make sure you add any missing trailing commas required for JSON:
+
+   ```json
+   {
+     "Kind": "service-defaults",
+     "Name": "static-client",
+     "Protocol": "http",
+     "MeshGateway": {},
+     "Expose": {},
+     "CreateIndex": 26,
+     "ModifyIndex": 26,
+     "Meta": { "consul.hashicorp.com/source-datacenter": "dc1" }
+   }
+   ```
+
+1. Write the updated config entry:
+   ```shell-session
+   $ consul config write static-client.json
+   Config entry written: service-defaults/static-client
+   ```
+1. Now you're ready to create a custom resource that takes over control of this
+   config entry. The custom resource will look like:
+
+   ```yaml
+   apiVersion: consul.hashicorp.com/v1alpha1
+   kind: ServiceDefaults
+   metadata:
+     name: static-client
+   spec:
+     protocol: 'http'
+   ```
+
+   Where `metadata.name` is the name of your service and `spec.protocol` is
+   the default protocol you've set.
+
+1. When you run `kubectl apply` on this file, the `ServiceDefaults` custom
+   resource should be created successfully and its `synced` status will be `True`:
+
+   ```shell-session
+   $ cat <<EOF | kubectl apply -f -
+   apiVersion: consul.hashicorp.com/v1alpha1
+   kind: ServiceDefaults
+   metadata:
+     name: static-client
+   spec:
+     protocol: "http"
+   EOF
+   servicedefaults.consul.hashicorp.com/static-client created
+
+   $ kubectl get servicedefaults static-client
+   NAME            SYNCED   AGE
+   static-client   True     1s
+   ```
+
+1. Repeat steps 2-6 for each service-defaults config entry.
+1. Finally, set `connectInject.centralConfig.defaultProtocol` to `""` and run a Helm upgrade to remind yourself that it is no longer used:
+   ```yaml
+   connectInject:
+     centralConfig:
+       defaultProtocol: ''
+   ```
```

Comment:
Does it makes sense to instead recommend deleting this section.
```suggestion
1. Finally, remove the `connectInject.centralConfig.defaultProtocol` entry and run a Helm upgrade for consistency:
   ```yaml
   connectInject:
     centralConfilg:
   ```
```

Reply:
> to remind yourself that it is no longer used

Not sure why but this phrasing really cracked me up

Reply:
^ditto 👍🏽